codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -226,6 +226,39 @@ class KotlinJmespathExpressionVisitor(
             VisitedExpression(ident)
         }
 
+        "abs" -> {
+            codegenReq(expression.arguments.size == 1) { "Unexpected number of arguments to $expression" }
+
+            val number = acceptSubexpression(expression.arguments[0])
```

Comment:
**Suggestion**: Now that we have more than two functions, a pattern begins to emerge in how we handle arguments. I think we could create extension functions which shorten/simplify getting the args based on arity:

```kotlin
fun ExpressionFunction.singleArg(): JmespathExpression {
    codegenReq(arguments.size == 1) { "Unexpected number of arguments to $this" }
    return acceptSubexpression(arguments[0])
}

fun ExpressionFunction.twoArgs(): Pair<JmespathExpression, JmespathExpression> {
    codegenReq(arguments.size == 2) { "Unexpected number of arguments to $this" }
    return acceptSubexpression(arguments[0]) to acceptSubexpression(arguments[1])
}
```

Then we could clean up the args parts of the `when` branches in `visitFunction`:

```kotlin
override fun visitFunction(expression: FunctionExpression): VisitedExpression = when (expression.name) {
    "contains" -> {
        val (subject, search) = expression.twoArgs()
        ...
    }

    "length" -> {
        val subject = expression.singleArg()
        ...
    }

    ...
}
```